src/integrationTest/java/org/opensearch/index/codec/rest/CreateIndexWithCodecIT.java

```diff
@@ -35,4 +64,68 @@ public void testCreateIndexWithZstdCodec() throws IOException {
 
         ensureGreen(index);
     }
+
+    @Override
+    protected RestClient buildClient(Settings settings, HttpHost[] hosts) throws IOException {
+        RestClientBuilder builder = RestClient.builder(hosts);
+        configureHttpOrHttpsClient(builder, settings);
+        builder.setStrictDeprecationMode(true);
+        return builder.build();
+    }
+    
+    protected void configureHttpOrHttpsClient(RestClientBuilder builder, Settings settings) throws IOException {
+        configureClient(builder, settings);
+        
+        if (getProtocol().equalsIgnoreCase("https")) {
+            final String username = System.getProperty("user");
+            if (Strings.isNullOrEmpty(username)) {
+                throw new RuntimeException("user name is missing");
+            }
+
+            final String password = System.getProperty("password");
```

Comment:
Maybe a non issue here since this is a simple test but my senses tingle when I see a password that doesn't use the closeable `SecureString` in a try-with-resources. I'm not sure there's a way to avoid returning the property as a `char[]` anyway so the String object allocation is likely unavoidable anyway.

Reply:
Thanks @nknize , this is sadly platform wide issue since the password is passed from CI to the tests (for all core and all pugins) using environment variables [1].

[1]  https://github.com/opensearch-project/opensearch-build/blob/main/scripts/default/integtest.sh#L98

Reply:
In a separate PR we should work to change that to use the `KeyStore`.

Reply:
:+1:  no objection, could you please open an issue [https://github.com/opensearch-project/opensearch-build](https://github.com/opensearch-project/opensearch-build/) here, I will provide more details if needed, thank you